src/sagemaker_containers/_mpi.py

```
@@ -0,0 +1,236 @@
+        self._master_hostname = str(master_hostname)
+
+    def run(self, wait=True, capture_error=False):  # type: (bool, bool) -> None
+        """The WorkerRunner proceeds as following:
+
+        - wait for the MPI Master to create its SSH daemon
+        - start its SSH daemon
+        - monitor the MPI orted process and wait it to finish the MPI execution
+        """
+        logger.info('Starting MPI run as worker node.')
+        if wait:
+            logger.info('Waiting for MPI Master to create SSH daemon.')
+            self._wait_master_to_start()
+        logger.info('MPI Master online, creating SSH daemon.')
+
+        _start_sshd_daemon()
+
+        if wait:
+            logger.info('Waiting for MPI process to finish.')
+            _wait_orted_process_to_finish()
+            time.sleep(30)
+        logger.info('MPI process finished.')
+
+    def _wait_master_to_start(self):  # type: () -> None
+        while not _can_connect(self._master_hostname):
+            time.sleep(1)
+
+    def _wait_master_to_finish(self):  # type: () -> None
+        while _can_connect(self._master_hostname):
+            time.sleep(30)
+
+
+def _wait_orted_process_to_finish():  # type: () -> None
+    orted = _orted_process()
+    psutil.wait_procs(orted)
+
+
+def _orted_process():
+    """Waits maximum of 5 minutes for orted process to start"""
+    for i in range(5 * 60):
+        procs = [p for p in psutil.process_iter(attrs=['name']) if p.info['name'] == 'orted']
+
+        if procs:
+            return procs
+        time.sleep(1)
+
+
+class MasterRunner(_process.ProcessRunner):
+    """Responsible to prepare MPI distributed training and syncronize work with the Workers.
+    """
+
+    def __init__(self, user_entry_point, args, env_vars, master_hostname, hosts, process_per_host,
+                 custom_mpi_options, network_interface_name, interval=1,
+                 timeout_in_seconds=60 * 60):
```

An hour seems huge for this timeout. For Chainer, workers would come up within seconds of each other, even for large clusters.

Could this be 1 minute? A shorter timeout benefits users since their jobs will fail faster (and cheaper).